### PR TITLE
Add proxy to redhat repository

### PR DESCRIPTION
**Pull request by elfelli**
Add proxy to redhat yum repository configuration file

# Description
To enable Proxy Settings for Yum configuration on a specifique repo like "confluent" we need to add the proxy configuration to the redhat yum repository delivered by this playbook.

Fixes # (issue)
Even with proxy_env settings enabled it's impossible to run yum install, we need to fix this task and add the proxy config on the redhat yum repo config

before setting proxy

```yml
- name: Add Confluent Dist Yum Repo
  yum_repository:
    name: Confluent.dist
    file: confluent
    description: "Confluent repository (dist)"
    baseurl: "{{confluent_common_repository_redhat_dist_baseurl}}"
    gpgcheck: "{{confluent_common_repository_redhat_dist_gpgcheck}}"
    gpgkey: "{{confluent_common_repository_redhat_dist_gpgkey}}"
    enabled: "{{confluent_common_repository_redhat_dist_enabled}}"
  register: confluent_dist_repo_result
  when:
    - repository_configuration == 'confluent'
    - installation_method == "package"
```

after setting proxy
```yml
- name: Add Confluent Dist Yum Repo
  yum_repository:
    name: Confluent.dist
    file: confluent
    description: "Confluent repository (dist)"
    baseurl: "{{confluent_common_repository_redhat_dist_baseurl}}"
    gpgcheck: "{{confluent_common_repository_redhat_dist_gpgcheck}}"
    gpgkey: "{{confluent_common_repository_redhat_dist_gpgkey}}"
    enabled: "{{confluent_common_repository_redhat_dist_enabled}}"
    proxy: "{{proxy_env.http_proxy | default(omit)}}"
  register: confluent_dist_repo_result
  when:
    - repository_configuration == 'confluent'
    - installation_method == "package"
```

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

molecule converge -s mtls-custombundle-rhel -- --skip-tags package

TASK [confluent.common : Check Internet Access for Confluent Packages/Archive] ***

ok: [kafka-rest1]
ok: [kafka-connect1]
ok: [zookeeper1]
ok: [ksql1]
ok: [kafka-broker1]
ok: [control-center1]
ok: [kafka-broker3]
ok: [kafka-broker2]
ok: [schema-registry1]

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

**Comment by nsharma-git** [comment]
Can you please update the target branch to any dev branch i.e 6.2.x

**Comment by elfelli** [comment]
Changing to dev branch

File: roles/confluent.common/tasks/redhat.yml [hunk 1]
```diff
@@ -22,6 +23,7 @@
     gpgcheck: "{{confluent_common_repository_redhat_main_gpgcheck}}"
     gpgkey: "{{confluent_common_repository_redhat_main_gpgkey}}"
     enabled: "{{confluent_common_repository_redhat_main_enabled}}"
+    proxy: "{{proxy_env.http_proxy}}"
```
Comment (nsharma-git) [inline roles/confluent.common/tasks/redhat.yml:26]: Do you want to use `environment` to make it consistent and usable with different protocols

File: roles/confluent.common/tasks/redhat.yml [hunk 2]
```diff
@@ -8,6 +8,7 @@
     gpgcheck: "{{confluent_common_repository_redhat_dist_gpgcheck}}"
     gpgkey: "{{confluent_common_repository_redhat_dist_gpgkey}}"
     enabled: "{{confluent_common_repository_redhat_dist_enabled}}"
+    proxy: "{{proxy_env.http_proxy}}"
```
Comment (domenicbove) [inline roles/confluent.common/tasks/redhat.yml:11]:
```suggestion
    proxy: "{{proxy_env.http_proxy | default(omit)}}"
```

File: roles/confluent.common/tasks/redhat.yml (hunk 1 above)
Comment (domenicbove) [inline roles/confluent.common/tasks/redhat.yml:26]:
```suggestion
    proxy: "{{proxy_env.http_proxy | default(omit)}}"
```

File: roles/confluent.common/tasks/redhat.yml (hunk 2 above)
Comment (domenicbove) [reply roles/confluent.common/tasks/redhat.yml:11]: can you test with `omit`? this is useful when the var is not defined

File: roles/confluent.common/tasks/redhat.yml (hunk 1 above)
Comment (domenicbove) [reply roles/confluent.common/tasks/redhat.yml:26]: i think this module write to a .repo file 

so the proxy argument will end up as a line in that file

https://docs.ansible.com/ansible/latest/collections/ansible/builtin/yum_repository_module.html

Comment (nsharma-git) [reply roles/confluent.common/tasks/redhat.yml]: Got it!

**Review by nsharma-git — APPROVED** [review]

**Review by domenicbove — CHANGES_REQUESTED** [review]
One more suggestion @elfelli but looks good. Can you update the description to show the contents of the repo file with the `proxy_env.http_proxy` env var set and not set?

File: roles/confluent.common/tasks/redhat.yml (hunk 2 above)
Comment (elfelli) [reply roles/confluent.common/tasks/redhat.yml:11]: > One more suggestion @elfelli but looks good. Can you update the description to show the contents of the repo file with the `proxy_env.http_proxy` env var set and not set?

done !

**Review by elfelli — COMMENTED** [review]
done !

File: roles/confluent.common/tasks/redhat.yml (hunk 2 above)
Comment (domenicbove) [reply roles/confluent.common/tasks/redhat.yml:11]: line 11 is missing the default omit

also i meant to ssh to a host and `cat /etc/yum.repos.d/confluent.repo`
and put that into the description, not copy the file changes

**Review by domenicbove — CHANGES_REQUESTED** [review]
One line needs updating

**Review by domenicbove — APPROVED** [review]
lgtm

**Pull request merged**